airbyte-integrations/connectors/source-okta/source_okta/schemas/user_role_assignments.json

```diff
@@ -1,3 +1,81 @@
 {
-  "$ref": "shared-role.json"
+  "type": ["null", "object"],
+  "properties": {
+    "id": {
+      "type": ["null", "string"]
```

Comment:
may I ask why `null` is an acceptable type for an id?

Reply:
If we receive somehow or accidentally `{"id": None }` the connector should be able to handle this properly, this is the common practice, does this touches your workflow in any way?

Reply:
I see, thanks for sharing. I saw it comes from an on-call issue, not sure if you are okay to share with it. 

I saw all properties are lifted to be a nullable type (or Optional<T> in Java). It greatly improves the compatibility, but I feel it's too "Duck Typing"? When an id, a primary key, is null in database, it can't be inserted. When a primary key is null in an incremental import, it will be ignored. 

I am curious more about the root cause, maybe you help capture a bug on the Okta side. I am happy to help debug it! 

Also, assuming the API or data source is problematic, and the schema blocks a new version release on Airbyte, you want to unblock it. Can the connectors platform offers a kill-switch to either skip tests, or temporarily uplift all fields to nullable? It's okay to defer the problem and fix it later. Overall, source-okta is not a big connector, but the PR contains a few hundred lines of change. I am just wondering if there's a more efficient way to address the hot-fix. 

BTW, I just added a new stream, #14739 and I did not use nullable in the new schema, it seems working from the test result. I am not sure if you want to apply a similar patch to the new schema.

Reply:
Please apply the same patch in your schemas, because data could differ from account to account, there is no confidence that some fields will always be not-null. Again, this is the best practice for connector development. 

> Can the connectors platform offers a kill-switch to either skip tests, or temporarily uplift all fields to nullable?
Not sure about that. @sherifnada

> Overall, source-okta is not a big connector, but the PR contains a few hundred lines of change. I am just wondering if there's a more efficient way to address the hot-fix.

Unfortunately, in this particular case - no.